std/os/linux/index.zig

```diff
@@ -624,6 +625,54 @@ pub const S_IWOTH = 0o002;
 pub const S_IXOTH = 0o001;
 pub const S_IRWXO = 0o007;
 
+pub const O_CREAT = 0o100;
```

Comment:
Although these constants are the same on x86_64 and aarch64, they are per-architecture values:

```
andy@xps ~/d/musl> grep -RI O_CREAT
arch/mips64/bits/fcntl.h:#define O_CREAT        0400
arch/generic/bits/fcntl.h:#define O_CREAT        0100
arch/x32/bits/fcntl.h:#define O_CREAT        0100
arch/powerpc64/bits/fcntl.h:#define O_CREAT        0100
arch/arm/bits/fcntl.h:#define O_CREAT        0100
arch/s390x/bits/fcntl.h:#define O_CREAT        0100
arch/mips/bits/fcntl.h:#define O_CREAT        0400
arch/x86_64/bits/fcntl.h:#define O_CREAT        0100
arch/aarch64/bits/fcntl.h:#define O_CREAT        0100
arch/m68k/bits/fcntl.h:#define O_CREAT        0100
arch/mipsn32/bits/fcntl.h:#define O_CREAT        0400
arch/powerpc/bits/fcntl.h:#define O_CREAT        0100
```

I think they should stay in the per-architecture files to prevent bugs when more architecture support is added in the future.